### ISSUE TITLE
Create a Homebrew formula for Notation CLI

### ISSUE THREAD

**Issue by FeynmanZhou** [notaryproject/notation#571]
### What is the areas you would like to add the new feature to?

Notation CLI

### Is your feature request related to a problem?

This feature request is derived from the Notary v2 roadmap, https://github.com/notaryproject/roadmap/issues/76.

To simplify the Notation CLI installation experience, we need to consider providing a native installation package on macOS or another Linux distro, that allows users to install Notation CLI via a simple command like `brew install notation`.

Uninstallation and upgrading are not confirmed in this program yet. We need to discuss if these two functionalities should be included in this .msi installer or not.

### What solution do you propose?

Write this brew formula using Ruby and add it to Homebrew-core. Follow the guideline of https://docs.brew.sh/Adding-Software-to-Homebrew to complete this submission. 

### What alternatives have you considered?

None

**Comment by FeynmanZhou** [comment]
Homebrew only allows accepting a stable release package and any alpha/RC release is not acceptable. So we still need to defer this implementation until a stable Notation CLI v1.0.0 release is available, which is mid of Apr.

**Comment by FeynmanZhou** [comment]
Hi @mintbomb27, we plan to cut the Notation v1.0.0 release around 1 week. Are you still interested in creating a Homebrew formula for Notation?

**Comment by mintbomb27** [comment]
Yes! Where shall I add it to? Shall I directly put a PR with `homebrew-core`? How shall I get it reviewed by you?

**Comment by FeynmanZhou** [comment]
@mintbomb27 Thanks! I think you will need to follow https://docs.brew.sh/Formula-Cookbook and submit a PR to add Notation v1.0.0 in https://github.com/Homebrew/homebrew-core

**Comment by mintbomb27** [comment]
@FeynmanZhou So I tried creating the formula for the rc7 release, and it will look like:
```rb
class Notation < Formula
  desc "CLI tool to sign and verify OCI artifacts and container images"
  homepage "https://notaryproject.dev/"
  url "https://github.com/notaryproject/notation/archive/refs/tags/v1.0.0-rc.7.tar.gz"
  sha256 "a7ea851b7bc85d65316ceda61ecb99b4cfb179d4feeb52f4ba1fcfd0784ccbcc"
  license "Apache-2.0"
  head "https://github.com/notaryproject/notation", branch: "main"

  depends_on "go" => :build

  def install
    ENV["CGO_ENABLED"] = "0"
    project = "github.com/notaryproject/notation"
    ldflags = %W[
      -s -w
      -X #{project}/internal/version.Version=v#{version} 
      -X #{project}/internal/version.BuildMetadata=
    ]
    system "go", "build", *std_go_args(output: bin/"notation", ldflags: ldflags), "./cmd/notation"
  end

  test do
    notation_version = shell_output("#{bin}/notation version | awk 'FNR==3{printf $2}'")
    assert_equal "v#{version}", notation_version
  end
end
```

In the `ldflags`, Is it necessary to add the `/internal/version.GitCommit` flag as well? I'll have to set the url to the git repo instead of the source tarball and get it from it then.

**Comment by iamsamirzon** [comment]
@yizha1  - Can you move it to another milestone after 1.0.0? This cannot be done in 1.0.0 because Hombrew does not accept RC releases

**Comment by yizha1** [comment]
@iamsamirzon which milestone do you suggest? Since it needs to be done right after 1.0.0 release, if we remove it, it seems not easy to be tracked.

**Comment by mintbomb27** [comment]
@FeynmanZhou @yizha1 Now that v1.0.0 is released, can you take a look at the above Homebrew formula? I'll go ahead and create the PR at Homebrew/homebrew-core soon.

**Comment by FeynmanZhou** [comment]
@mintbomb27 Thanks for your support. Could you pls help raise a PR to add Notation at [Homebrew/homebrew-core](https://github.com/Homebrew/homebrew-core)?

**Comment by mintbomb27** [comment]
Done at https://github.com/Homebrew/homebrew-core/pull/139873

**Comment by mintbomb27** [comment]
@FeynmanZhou @shizhMSFT @yizha1 Would be great if anyone could approve the PR there so that it could be merged.

**Comment by shizhMSFT** [comment]
> @FeynmanZhou @shizhMSFT @yizha1 Would be great if anyone could approve the PR there so that it could be merged.

@mintbomb27 I've reviewed the PR again and left some comments there.

**Comment by shizhMSFT** [comment]
@mintbomb27 The output of `notation version` is not desired as it prints
```console
# notation version
notation version
Notation - a tool to sign and verify artifacts.

Version:     v1.0.0+Homebrew
Go version:  go1.21.0
Git commit:  Homebrew
```
Could you follow https://github.com/Homebrew/homebrew-core/pull/139873#discussion_r1303151618 and send out a PR to fix it?

**Comment by mintbomb27** [comment]
Yes, I didn't make that change. @chenrui333 did. But what was the problem with the `Utils.git_short_head`? It actually had the git commit right?

**Comment by yizha1** [comment]
@FeynmanZhou Can we close this issue and create a new issue to track the problem mentioned by Shiwei

**Comment by mintbomb27** [comment]
Sure I can take it up.
